flow/src/main/java/flow/Flow.java

```diff
@@ -305,6 +327,14 @@ void enqueue(PendingTraversal pendingTraversal) {
       }
     }
 
+    void bootstrap(History history) {
+      this.nextHistory = checkNotNull(history, "history");
+      if (dispatcher == null) {
+        throw new AssertionError("Bad doExecute method allowed dispatcher to be cleared");
+      }
+      dispatcher.dispatch(new Traversal(null, nextHistory, Direction.REPLACE), this);
```

Comment:
It's your call, but I'll make a bet right now that this causes NPEs and otherwise unnecessary defensive code.